robottelo/test.py

```diff
@@ -211,6 +232,128 @@ def setUp(self):
         self.user = User(self.browser)
         self.usergroup = UserGroup(self.browser)
 
+    def take_screenshot(self, testmethodname):
+        """Takes screenshot of the UI and saves it to the disk by creating
+        a directory same as that of the test method name."""
+        fmt = '%Y-%m-%d_%H:%M:%S'
+        filename = datetime.now().strftime(fmt)
+        # testmethodname varies so trying to handle various scenarios.
+        # pop depending on ddt, normal test or ddt's 1st job.
+        # For ddt tests with maps.
+        if "___" in testmethodname:
+            parts = testmethodname.split("___")
+        # For ddt tests with utf8, html, etc.
+        elif "__" in testmethodname:
+            parts = testmethodname.split("__")
+        # For ddt test with alpha, numeric, etc.
+        else:
+            testmethodname = testmethodname.split('_')
+            for _ in range(2):
+                testmethodname.pop()
+        if "___" in testmethodname or "__" in testmethodname:
+            testmethodname = parts[0]
+            testmethodname = testmethodname.split('_')
+            testmethodname.pop()
+        testmethodname = "_".join(testmethodname)
```

Comment:
The issue here is when we the tests are "Data driven". As then instead of the just the `testMethodName` even the `data` along with it's position in the list is mentioned.
And hence all the fiddling with the string.

For example: 
a) No need to change the string name for **Non DDT**  tests as it is only, 
`test_positive_create_activation_key_1`
b) For **DDT* tests, the value of `self._testMethodName` appear as `test_positive_create_activation_key_1_5__xd0_xfe_xdb_xf1_xf9_xe6_xf3_xfb`